pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -436,10 +622,31 @@ func generatePeerConfigs(subnets []string, gatewayIP net.IP) (map[*net.IPNet]net
 	return peerConfigs, nil
 }
 
-// getPeerGatewayIP will always return the first Gateway IP.
-func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo) net.IP {
+// getPeerGatewayIP will return the network IP address of the Service CIDR when WireGuard enabled,
+// or it will return the first IP of the Gateway.
+func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo, enableWireGuard bool) net.IP {
+	if enableWireGuard {
+		_, serviceCIDR, err := net.ParseCIDR(spec.ServiceCIDR)
+		if err != nil {
+			klog.Warningf("Failed to get the peer Gateway IP for WireGuard, error: %v", err)
+			return nil
+		}
+		return serviceCIDR.IP.Mask(serviceCIDR.Mask)
+	}
 	if len(spec.GatewayInfos) == 0 {
 		return nil
 	}
 	return net.ParseIP(spec.GatewayInfos[0].GatewayIP)
 }
+
+func getLocalGatewayIP(gateway *mcv1alpha1.Gateway) net.IP {
+	if gateway.WireGuard != nil {
```

Comment:
I feel you should use the same condition as getPeerGatewayIP to check `c.wireGuardConfig != nil`.

Reply:
Done. Using the same condition now.